modules/storage/manager.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package storage
+
+import (
+	"encoding/json"
+	"sort"
+	"sync"
+)
+
+var manager *Manager
+
+// Manager is the storage manager
+type Manager struct {
+	mutex sync.Mutex
```

Comment:
Better is to use RWMutex

Reply:
Fair enough, the mutex is held for only the shortest of times - that's why I didn't bother with the rwmutex but I will switch to it